src/core/handlers/GraphQLHandler.test.ts

```diff
@@ -322,6 +370,161 @@ describe('parse', () => {
       })
     })
   })
+
+  describe('with endpoint configuration', () => {
+    test('parses a request when the graphql.link endpoint matches', async () => {
+      const handler = new GraphQLHandler(
+        OperationTypeNode.QUERY,
+        'GetUser',
+        'https://mswjs.com/graphql',
+        resolver,
+      )
+
+      await expect(
+        handler.parse({
+          request: createGetGraphQLRequest(
+            {
+              query: GET_USER,
+              variables: {
+                userId: 'abc-123',
+              },
+            },
+            'https://mswjs.com/graphql',
+          ),
+        }),
+      ).resolves.toEqual({
+        match: {
+          matches: true,
+          params: {},
+        },
+        operationType: 'query',
+        operationName: 'GetUser',
+        query: GET_USER,
+        variables: {
+          userId: 'abc-123',
+        },
+      })
+
+      await expect(
+        handler.parse({
+          request: createPostGraphQLRequest(
+            {
+              query: GET_USER,
+              variables: {
+                userId: 'abc-123',
+              },
+            },
+            'https://mswjs.com/graphql',
+          ),
+        }),
+      ).resolves.toEqual({
+        match: {
+          matches: true,
+          params: {},
+        },
+        operationType: 'query',
+        operationName: 'GetUser',
+        query: GET_USER,
+        variables: {
+          userId: 'abc-123',
+        },
+      })
+    })
+
+    test('parses a request when the graphql.link hostname does not match', async () => {
```

Comment:
Shouldn't this say "does not part the request" since the parsing is effectively skipped (no `query`, `variables`, etc. produced)?

Reply:
The handler.parse succeeds successfully, but we don't parse the request body out. 

I'll work on the language here to make that a bit more clear.

Reply:
https://github.com/mswjs/msw/pull/1871/commits/064cfe1ed02b96583526a5c60592a98ede0914d4

updated. maybe a bit verbose now, but hopefully expresses that well.  I wanted to avoid it reading like `handler.parse` failed or returned nothing, when it's mostly focused on not returning the gql stuff from the request

Reply:
Thank you!